echopype/convert/parsed_to_zarr_ek60.py

```diff
@@ -0,0 +1,272 @@
+import numpy as np
+import pandas as pd
+import psutil
+
+from .parsed_to_zarr import Parsed2Zarr
+
+
+class Parsed2ZarrEK60(Parsed2Zarr):
+    """
+    Facilitates the writing of parsed data to
+    a zarr file for the EK60 sensor.
+    """
+
+    def __init__(self, parser_obj):
+        super().__init__(parser_obj)
+
+        self.power_dims = ["timestamp", "channel"]
+        self.angle_dims = ["timestamp", "channel"]
+        self.p2z_ch_ids = {}  # channel ids for power, angle, complex
+        self.datagram_df = None  # df created from zarr variables
+
+    @staticmethod
+    def _get_string_dtype(pd_series: pd.Index) -> str:
+        """
+        Returns the string dtype in a format that
+        works for zarr.
+
+        Parameters
+        ----------
+        pd_series: pd.Index
+            A series where all of the elements are strings
+        """
+
+        if all(pd_series.map(type) == str):
+            max_len = pd_series.map(len).max()
+            dtype = f"<U{max_len}"
+        else:
+            raise ValueError("All elements of pd_series must be strings!")
+
+        return dtype
+
+    def _write_power(self, df: pd.DataFrame, max_mb: int) -> None:
+        """
+        Writes the power data and associated indices
+        to a zarr group.
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            DataFrame that contains power data
+        max_mb : int
+            Maximum MB allowed for each chunk
+        """
+
+        # obtain power data
+        power_series = df.set_index(self.power_dims)["power"].copy()
+
+        # get unique indices
+        times = power_series.index.get_level_values(0).unique()
+        channels = power_series.index.get_level_values(1).unique()
+
+        self.p2z_ch_ids["power"] = channels.values  # store channel ids for variable
+
+        # create multi index using the product of the unique dims
+        unique_dims = [times, channels]
+
+        power_series = self.set_multi_index(power_series, unique_dims)
+
+        # write power data to the power group
+        zarr_grp = self.zarr_root.create_group("power")
+        self.write_df_column(
+            pd_series=power_series,
+            zarr_grp=zarr_grp,
+            is_array=True,
+            unique_time_ind=times,
+            max_mb=max_mb,
+        )
+
+        # write the unique indices to the power group
+        zarr_grp.array(
+            name=self.power_dims[0], data=times.values, dtype="<M8[ns]", fill_value="NaT"
```

Comment:
You hardwired the dtype here. Is it more appropriate to just call `times.dtype` instead?

Reply:
I hardwired the dtype here because I was being overly cautious. I do not see the harm in using the more general solution you highlight, in fact, it is more appropriate. One small note, it looks like zarr requires this to be a string, so I will instead replace this with `times.dtype.str`.